### ISSUE TITLE
Rule for semis on non-object types

### ISSUE THREAD

**Issue by park-jemin** [eslint-stylistic/eslint-stylistic#233]
### Clear and concise description of the problem

Is there a rule or configuration for enforcing consistent semis in one-line types that are _not_ objects? 

### Suggested solution

Something like:

```ts
// valid
type Hash = Record<string, string>
type TimeOfDay = 'Morning' | 'Afternoon' | 'Evening' | 'Night'

// invalid
type Hash = Record<string, string>;
type TimeOfDay = 'Morning' | 'Afternoon' | 'Evening' | 'Night';

// or vice versa!
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [ ] If this feature request is accepted, I am willing to submit a PR to fix this issue

**Comment by antfu** [comment]
I don't understand why you would need that. But for niche requirements, I'd generally suggest you write your own ad-hoc rule to fix that

**Comment by park-jemin** [comment]
It is minor, but I think it's in the same realm as [`member-delimiter-style`](https://eslint.style/rules/ts/member-delimiter-style) - useful for allowing more flexibility in consistency across type definitions where something like Prettier would just shotgun them all. Definitely something I think belongs in the stylistic ruleset. 

My team doesn't use the typescript `semi` rule because of this at least. If we have the `ts/semi` rule on, it would create a bit of inconsistency in the object types/interfaces:

```ts
type ErrorResponse = string | { 
  code: string
  error: string
}; // ts/semi would force this semi here

interface SuccessResponse {
  code: string
  msg: string
}  // but not here
```

**Comment by antfu** [comment]
Yes it's in the scope of stylistic rules for sure. But the thing is that we can't possibly support every needs in the world and we need to keep the options interface in a maintainable scope. You haven't explain why this convention make senses (to me personally it's inconsistent)
